### PR TITLE
10 implement signin signup deleteuser

### LINKED ISSUE #10: Implement SignIn / SignUp / DeleteUser
**Short description:**
Add SignIn, SignUp and DeleteUser on back side.

**Describe the solution you'd like**
Create authentication system, by signing up with user and email. This will create a new user in the database.
Then, a user will be able to sign in on an account with user and email as well.
When DeleteUser, the user will be removed from the database.
Please be use Passport plugin.

### PR DESCRIPTION
**Pull request by LucasTesnier**
# Purpose:
The purpose of this pull request is to add the authentification process by using Passport.

## Added features:
- This service has been added to the application. It's include:
    - A signup path to create a new user in the database
    - A login path to connect with an existing user
    - A profile path which is only accessible with a bearer token
    - A bearer token implementation with time, delete, creation and link

## Updated features:
    - A refactoring of the server project architecture 
    - A reduction of the waiting time indeed by the database image (Now, 5 seconds of waiting but 40 tries)
    - Database.env file have been updated (Don't worry about the key it's a false one for unit testing)

## Removed features:
- Removed all the demonstrative path inside the server_app.js file

### CODE REVIEW

File: server/server_app.js [hunk 1]
```diff
@@ -72,87 +68,69 @@ app.get('/about.json', (req, res) => {
...
-  } catch (err) {
-    res.status(500).send(err)
+/**
+ * Get request accessing to the user profile.
+ * Need to be authentified with a token.
+ */
+app.get(
+  '/profile',
+  passport.authenticate('jwt', { session: false }),
+  function (req, res) {
```
Comment (tbellicha) [inline server/server_app.js]: Use `=>` instead of `function`

File: server/passport/token.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+const options = {
+  secretOrKey: process.env.JWT_SECRET,
+  jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken()
+}
+
+/**
+ * Setup a passport strategy option for Bearer Token
+ */
+passport.use(
+  new Strategy(options, async function (jwt_payload, done) {
```
Comment (tbellicha) [inline server/passport/token.js]: Use `=>` instead of `function`

**Review by GuyomT — DISMISSED** [review]
NTR

**Review by tbellicha — CHANGES_REQUESTED** [review]

File: server/server_app.js (hunk 1 above)
Comment (LucasTesnier) [reply server/server_app.js]: done :)

File: server/passport/token.js (hunk 2 above)
Comment (LucasTesnier) [reply server/passport/token.js]: done :)

**Review by tbellicha — APPROVED** [review]

**Review by ZQUEMA — APPROVED** [review]

**Pull request merged**